userguide/tutorials/payment_plugin.adoc

```diff
@@ -79,34 +72,33 @@ The following table lists the methods in https://github.com/killbill/killbill-pl
 
 * It is quite possible that your plugin might not support some of the operations listed above. In such a case, the plugin can do one of the following:
 
-** Return an *empty list* when the return type is a list 
-** Return a transaction with status *CANCELED* if the return type is a `PaymentTransactionInfoPlugin` 
-** Return empty objects otherwise
+** Return an *empty list* when the return type is a `List`.
+** Return a transaction with status *CANCELED* if the return type is a `PaymentTransactionInfoPlugin` .
+** Return empty objects otherwise.
 
 [[PaymentTransactionInfoPlugin]]
 == PaymentTransactionInfoPlugin 
 
+Most of the methods in the https://github.com/killbill/killbill-plugin-api/blob/master/payment/src/main/java/org/killbill/billing/payment/plugin/api/PaymentPluginApi.java[_PaymentPluginApi_] interface return a https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/plugin/api/PaymentTransactionInfoPlugin.java[_PaymentTransactionInfoPlugin_] object. Again, the https://github.com/killbill/killbill-plugin-framework-java[_Kill Bill Plugin framework_] includes a class called https://github.com/killbill/killbill-plugin-framework-java/blob/46d94fbeb1cf089aa04e62cfecf751ca47032023/src/main/java/org/killbill/billing/plugin/api/payment/PluginPaymentTransactionInfoPlugin.java[ _PluginPaymentTransactionInfoPlugin_] which implements the `PaymentTransactionInfoPlugin` interface. Thus, if you use the framework, you can simply create a class that extends `PluginPaymentTransactionInfoPlugin` and have the plugin methods return an object of this class.
 
-Most of the methods in the `PaymentPluginApi` return a https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/plugin/api/PaymentTransactionInfoPlugin.java[PaymentTransactionInfoPlugin] object. Again, the Kill Bill Plugin framework includes a class called https://github.com/killbill/killbill-plugin-framework-java/blob/46d94fbeb1cf089aa04e62cfecf751ca47032023/src/main/java/org/killbill/billing/plugin/api/payment/PluginPaymentTransactionInfoPlugin.java[ PluginPaymentTransactionInfoPlugin] which implements the `PaymentTransactionInfoPlugin`. You can create a class that extends this class and have the plugin methods return this object.
-
-The `PaymentTransactionInfoPlugin` has the following methods (Your plugin code needs to populate this information in the `PaymentTransactionInfoPlugin` object and a client application can then invoke these methods to retrieve this information):
+The `PaymentTransactionInfoPlugin` interface has the following methods (Your plugin code needs to populate this information in the `PaymentTransactionInfoPlugin` object):
```

Comment:
```suggestion
The `PaymentTransactionInfoPlugin` interface has the following methods, and your plugin code needs to populate this information in the `PaymentTransactionInfoPlugin` object:
```

Reply:
^^A suggestion to avoid using the parentheses and important information.

Reply:
Removed parentheses but kept two separate sentences.